packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1087,6 +1123,20 @@ class SliverAppBar extends StatefulWidget {
   /// Defaults to [NavigationToolbar.kMiddleSpacing].
   final double titleSpacing;
 
+  /// The height used for the [title].
+  ///
+  /// This value must not be null, and defaults to [kToolbarHeight].
+  final double titleHeight;
+
+  /// Whether [leading] and [actions] should always be placed vertically center.
+  /// If [titleHeight] is not modified, this option has NO effect.
+  ///
+  /// If true, [leading] and [actions] are always placed vertically center.
+  /// If false, icons will be placed on top.
+  ///
+  /// Defaults to true.
+  final bool centerIcons;
```

Comment:
I think we should hold off on this and wait for someone to ask for it.

I'm having trouble imagining why you would want a title bar where the actions were not centered along with the text. We should generally avoid adding API until we're sure we need it.

Reply:
(And in fact, I'm not sure a simple bool even makes the most sense here - we'd probably want some kind of alignment based property instead if we did this)